### PR TITLE
Update Commands

### PR DESCRIPTION
**Pull request by mosemister**
Adds commands back to the docs

Updates all the command docs with Raw and Parameter. I have skipped Flags but this is because I havent worked with them before on API 8. Plan to do that later

### CODE REVIEW

File: source/plugin/commands/argumentparsing.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+Minecraft's Brigadier system includes a powerful argument parser that Sponge can take advantage of. 
+It converts the string input to java base types (integer, booleans, string) or game objects 
+(players, worlds, block types, ...). Sponge's system also supports optional arguments and flags. It also handles 
+suggestions of arguments.
+
+The parsed arguments are stored in the :javadoc:`CommandContext` object. If the parser returns a single object, 
+obtain it with :javadoc:`CommandContext#one(ResourceKey)`. Optional and weak arguments may return ``Optional.empty()``
+
+Many of the parsers may return more than one object. Example multiple players with a matching username. In that case,
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: This should be one sentence, i.e.
`Many of the parsers may return more than one object; for example, multiple players [...]`

File: source/plugin/commands/argumentparsing.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+The parsed arguments are stored in the :javadoc:`CommandContext` object. If the parser returns a single object, 
+obtain it with :javadoc:`CommandContext#one(ResourceKey)`. Optional and weak arguments may return ``Optional.empty()``
+
+Many of the parsers may return more than one object. Example multiple players with a matching username. In that case, 
+you must use the :javadoc:`CommandContext#all(ResourceKey)` method to get the ``Collection`` of possible matches. 
+*Otherwise, the context object will throw an exception!*
+
+When creating a command, consider whether the argument could return multiple values, for example whether a player 
+argument could support multiple players when using a selector. If you support multiple values the users need to type 
+only one command and can use an easier command sytax. Example ``/tell @a Who took the cookies?``
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: sytax -> syntax
More commas needed, and the example can end the sentence - i.e.
`If you support multiple values, the users need to type only one command, and can use an easier command syntax, e.g. ``/tell @a [...]`

File: source/plugin/commands/argumentparsing.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+To create a new :javadoc:`Parameter` (argument), use the :javadoc:`Parameter` class that will give you many 
+:javadoc:`Parameter.Builder` options. Each parameter will need its :javadoc:`Parameter.Builder#key(String)` 
+filled out before being built. 
+
+Apply the ``Parameter`` to the command builder with the :javadoc:`Command.Builder#addParameter` method. 
+It is possible to pass more than one ``Parameter`` to the method, thus chaining multiple arguemtns.
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: arguments

File: source/plugin/commands/argumentparsing.rst [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                String message = context.one(messageParameter);
+
+                player.sendMessage(Component.text(message));
+                return CommandResult.success();
+            })
+            .addParameter(playerParameter, messageParameter)
+            .build();
+    }
+
+Overview of the ``Parameter`` command elements
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: Title Case please (Command Elements)

File: source/plugin/commands/argumentparsing.rst [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      - inherited
+
+.. tip::
+    See the Javadocs for :javadoc:`Parameter` for more information
+
+.. warning::
+
+    Don't expect that a ``Parameter`` will only ever return a single value, 
+    a lot of them support multiple return values; some might support regular expressions or use command selector. 
+    This is intensional as it makes commands easier to use. Example ``/tell @a BanditPlayer has cookies!``.
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: intensional -> intentional 
Make the example the end of the sentence (`[...] to use, e.g. ``/tell [...]`)

File: source/plugin/commands/argumentparsing.rst [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    
+Custom Parameter
+~~~~~~~~~~~~~~~~
+
+It is possible to create custom command elements. Example Vector2i. This is done though the
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: `It is possible to create custom command elements; for example, Vector2i`

File: source/plugin/commands/argumentparsing.rst [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+    Don't expect that a ``Parameter`` will only ever return a single value, 
+    a lot of them support multiple return values; some might support regular expressions or use command selector. 
+    This is intensional as it makes commands easier to use. Example ``/tell @a BanditPlayer has cookies!``.
+    
+Custom Parameter
+~~~~~~~~~~~~~~~~
+
+It is possible to create custom command elements. Example Vector2i. This is done though the 
+:javadoc:`Parameter#builder(Class)` method, which returns a :javadoc:`Parameter.Builder` where all data of the parameter
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: "where all data of the parameter is needed to be provided." is a bit hard to parse.
Do you mean "where all the data the parameter needs should be provided." ?

File: source/plugin/commands/argumentparsing.rst [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+is needed to be provided. Once done call the ``build`` method to build the parameter.
+
+When building a new parameter, only the ``parser`` and ``key`` are required for build. The ``parser`` is where the logic
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: "for build" at the end seems redundant.
The meaning of the next is a bit unclear. Should it be more like 
"The ``parser`` contains the logic for mapping the ``String`` input to the desired value." ?

File: source/plugin/commands/argumentparsing.rst [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+          int x = reader.parseInt();
+          int y = reader.parseInt();
+          return new Vector2i(x, y);
+      })
+      .key("vector")
+      .build();
+
+.. tip::
+
+    When building a new parameter, you can base your parameter from a exsiting parameter. Example: taking
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: "you can base your parameter on an existing parameter".
i.e. from -> on 
and fix typo

File: source/plugin/commands/argumentparsing.rst [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+          int y = reader.parseInt();
+          return new Vector2i(x, y);
+      })
+      .key("vector")
+      .build();
+
+.. tip::
+
+    When building a new parameter, you can base your parameter from a exsiting parameter. Example: taking 
+    :javadoc:`Parameter#string()` and giving it client suggestions
```
Comment (Inscrutable) [inline source/plugin/commands/argumentparsing.rst]: suggestions.

File: source/plugin/commands/childcommands.rst [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    org.spongepowered.api.command.Command.CommandExecutor
+    org.spongepowered.api.command.exception.ArgumentParseException
+
+The :javadoc:`Command.Builder` supports hierachical command structure like this:
+
+- ``/mail`` (parent command)
+    - ``/main send`` (child command)
+    - ``/main read`` (child command)
+
+Every child command is a seperate ``Command`` and can be cheated in the same way as a regular command is.
```
Comment (Inscrutable) [inline source/plugin/commands/childcommands.rst]: cheated -> created
(at least, I hope that's what you meant)

File: source/plugin/commands/childcommands.rst [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+        .child(readCmd, "read", "r", "inbox")
+        .child(sendCmd, "send", "s", "write")
+        .build();
+
+Fallback Behavior
+=================
+
+If a command has child commands, a :javadoc:`CommandExecutor`, set through 
+:javadoc:`Command.Builder#executor(CommandExecutor)` and the parameters for the executor are set to optional. 
+The behavior of error in selection and argument parsing of child commands is dependent on whether this parent
```
Comment (Inscrutable) [inline source/plugin/commands/childcommands.rst]: It might be better to rephrase this, it's a bit opaque. Is this about "errors in selection and argument parsing of child commands", or are they two separate things? If the latter, at least separate them with commas. Similarly, we could perhaps make this clearer by replacing "The behavior of" with "What happens with"

File: source/plugin/commands/childcommands.rst [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+Fallback Behavior
+=================
+
+If a command has child commands, a :javadoc:`CommandExecutor`, set through 
+:javadoc:`Command.Builder#executor(CommandExecutor)` and the parameters for the executor are set to optional. 
+The behavior of error in selection and argument parsing of child commands is dependent on whether this parent 
+executor exists.
+
+If a parent exector has not been set, then if the requested child command could not be found or the arguments cannot
```
Comment (Inscrutable) [inline source/plugin/commands/childcommands.rst]: change this first `then` into an `and`

File: source/plugin/commands/childcommands.rst [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+:javadoc:`Command.Builder#executor(CommandExecutor)` and the parameters for the executor are set to optional. 
+The behavior of error in selection and argument parsing of child commands is dependent on whether this parent 
+executor exists.
+
+If a parent exector has not been set, then if the requested child command could not be found or the arguments cannot 
+be parsed, then an :javadoc:`ArgumentParseException` will be thrown.
+
+.. warning::
+
+    If a ``ArgumentParseException`` is unhandled, then the Minecraft brigader system will tell the user that the
```
Comment (Inscrutable) [inline source/plugin/commands/childcommands.rst]: a -> an
unhandled -> not handled
brigader -> Brigadier

Much as I like people inventing new words when we need them...

File: source/plugin/commands/commandbuilding.rst [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+    org.spongepowered.api.command.CommandExecutor
+    org.spongepowered.api.command.parameter.CommandContext
+    org.spongepowered.api.command.CommandResult
+    org.spongepowered.api.command.exception.CommandException
+    org.spongepowered.api.command.exception.ArgumentParseException
+    org.spongepowered.api.event.lifecycle.RegisterCommandEvent
+    org.spongepowered.api.event.EventContextKeys
+    
+The first step is to get a new :javadoc:`Command.Builder` builder. 
+The builder provides methods to modify the command help messages, command arguments and the command logic.
```
Comment (Inscrutable) [inline source/plugin/commands/commandbuilding.rst]: comma after arguments

File: source/plugin/commands/commandbuilding.rst [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+    * - children
+      - Sets the child commands of this command with their aliases
+    * - executionRequirements
+      - Sets the execution requirements for the command.
+    * - terminal
+      - Sets if the command can execute despite the parameters stating that they are mandatory
+    * - build
+      - Builds the command. After that you need to register the command
+
+Player-Only commands
```
Comment (Inscrutable) [inline source/plugin/commands/commandbuilding.rst]: Title Case

File: source/plugin/commands/commandbuilding.rst [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+~~~~~~~~~~~~~~~~~~~~
+
+Sometimes it is necessary that only players can execute a command. Example ``/suicide`` command.
```
Comment (Inscrutable) [inline source/plugin/commands/commandbuilding.rst:74]: Make into one sentence. `execute a command, for example, "/suicide".`

File: source/plugin/commands/commandbuilding.rst [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+            .build();
+    }
+
+.. note::
+
+    We recommend you to add an optional ``[player]`` argument to make the command *console-friendly*. 
+    Example ``/suicide [player]``
+
+    The easiest solution for this is to append a ``playerOrTarget`` command parameter (see :doc:`argumentparsing`) 
+    instead of providing a execution requirement.
```
Comment (Inscrutable) [inline source/plugin/commands/commandbuilding.rst]: a -> an

File: source/plugin/commands/commandbuilding.rst [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+
+.. tip::
+
+    Often times command are put in as player-only as they require the location in the world the command was
+    executed from. Best practise would be to check for the target location using :javadoc:`EventContextKeys#LOCATION`
+    instead of the player as this would allow command blocks to run the command without specifing a player.
```
Comment (Inscrutable) [inline source/plugin/commands/commandbuilding.rst]: add a comma after "the player"
specifing -> specifying

File: source/plugin/commands/commandbuilding.rst [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+.. tip::
+
+    You can use `anonymous classes <https://docs.oracle.com/javase/tutorial/java/javaOO/anonymousclasses.html>`_ to 
+    define the command executor in the command's build process (see example in the :doc:`argumentparsing` page).
+
+Command Result
+~~~~~~~~~~~~~~
+
+The command result is used to give more information about how the command was executed.
+In the example above we used :javadoc:`CommandResult#success` however we can give more information then this,
```
Comment (Inscrutable) [inline source/plugin/commands/commandbuilding.rst]: comma needed after the javadoc reference
then -> than

File: source/plugin/commands/commandbuilding.rst [hunk 21]
```diff
@@ -0,0 +1,200 @@
...
+which is then sent back to the client to provide a correctly formatted message.
+
+By going the builder route you gain access to the ``result`` method which is a integer value sent back to the client.
```
Comment (Inscrutable) [inline source/plugin/commands/commandbuilding.rst]: a -> an

File: source/plugin/commands/commandbuilding.rst [hunk 22]
```diff
@@ -0,0 +1,200 @@
...
+    * - Value
+      - Description
+    * - Positive
+      - Successful execution
+    * - Zero
+      - Unsuccessful execution (but not necessarily an error)
+    * - Negative
+      - Undefined in the Minecraft spec, can result in different effects
+
+Error handling
```
Comment (Inscrutable) [inline source/plugin/commands/commandbuilding.rst]: Title Case

File: source/plugin/commands/rawcommand.rst [hunk 23]
```diff
@@ -0,0 +1,78 @@
...
+
+.. javadoc-import::
+    org.spongepowered.api.event.lifecycle.RegisterCommandEvent
+    org.spongepowered.api.command.Command.Raw
+
+The raw command interface can be used to define commands. The interface can be used as a base for custom command APIs.
+
+It is recommended to use the :doc:`commandbuilding` for simple command definitions.
+
+Writing a command
```
Comment (Inscrutable) [inline source/plugin/commands/rawcommand.rst]: Title Case

File: source/plugin/commands/rawcommand.rst [hunk 24]
```diff
@@ -0,0 +1,78 @@
...
+        public Component usage(CommandCause cause) {
+            return this.usage;
+        }
+    }
+
+.. tip::
+
+    See the JavaDoc for :javadoc:`Command.Raw` for the purposes of each method in the example
+
+Registering the command
```
Comment (Inscrutable) [inline source/plugin/commands/rawcommand.rst]: Title Case

**Review by Inscrutable — COMMENTED** [review]
Another herculean effort, thanks!
Some bits need clarifying, and there's the usual collection of typos and absent Title Case stuff.

File: source/plugin/commands/rawcommand.rst (hunk 24 above)
Comment (mosemister) [reply source/plugin/commands/rawcommand.rst]: Updated

File: source/plugin/commands/rawcommand.rst (hunk 23 above)
Comment (mosemister) [reply source/plugin/commands/rawcommand.rst]: Updated

File: source/plugin/commands/commandbuilding.rst (hunk 22 above)
Comment (mosemister) [reply source/plugin/commands/commandbuilding.rst]: Updated

File: source/plugin/commands/commandbuilding.rst (hunk 21 above)
Comment (mosemister) [reply source/plugin/commands/commandbuilding.rst]: Updated

File: source/plugin/commands/commandbuilding.rst (hunk 20 above)
Comment (mosemister) [reply source/plugin/commands/commandbuilding.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 5 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 4 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 3 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 2 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 1 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst [hunk 25]
```diff
@@ -0,0 +1,176 @@
+================
+Argument Parsing
+================
+
+.. javadoc-import::
+    org.spongepowered.api.command.parameter.CommandContext
+    org.spongepowered.api.command.parameter.Parameter
+    org.spongepowered.api.command.Parameter.Parameter.Builder
```
Comment (ST-DDT) [inline source/plugin/commands/argumentparsing.rst]:
```suggestion
    org.spongepowered.api.command.parameter.Parameter.Builder
```

Comment (ST-DDT) [reply source/plugin/commands/argumentparsing.rst]: Also the import doesn't seem to work.

**Review by ST-DDT — COMMENTED** [review]
Please check the imports/javadocs-references again:

````txt
Processing: plugin/commands/argumentparsing.rst
 - Unknown import "Parameter.Builder" in org.spongepowered.api.command.parameter.Parameter.Builder
 - Unknown import "ResourceKey" in CommandContext#one(ResourceKey)
 - Unknown import "ResourceKey" in CommandContext#all(ResourceKey)
 - Missing import "Parameter.Builder" in Parameter.Builder
 - Missing import "Parameter.Builder" in Parameter.Builder#key(String)
 - Unknown field "addParameter" in Command.Builder#addParameter
 - Unknown import "Class" in Parameter#builder(Class)
 - Missing import "Parameter.Builder" in Parameter.Builder
Processing: plugin/commands/childcommands.rst
 - Unknown import "Command.CommandExecutor" in org.spongepowered.api.command.Command.CommandExecutor
 - Unknown import "Command" in Command.Builder#child(Command,String,String...)
 - Unknown import "String" in Command.Builder#child(Command,String,String...)
 - Unknown import "String..." in Command.Builder#child(Command,String,String...)
 - Missing import "CommandExecutor" in CommandExecutor
 - Unknown import "CommandExecutor" in Command.Builder#executor(CommandExecutor)
Processing: plugin/commands/commandbuilding.rst
 - Import verification error "EventContextKeys" in org.spongepowered.api.event.EventContextKeys
 - Unknown import "Builder` builder." in The first step is to get a new :javadoc:`Command.Builder` builder.
 - Unknown import "" in The builder provides methods to modify the command help messages, command arguments and the command logic.
 - Unknown import "" in These methods can be chained.
 - Unknown field "build" in Command.Builder#build
 - Missing import "EventContextKeys" in EventContextKeys#LOCATION
 - Unknown field "success" in CommandResult#success
 - Unknown field "success" in CommandResult#success
 - Unknown import "Component" in CommandResult#error(Component)
 - Unknown import "PluginContainer" in RegisterCommandEvent#register(PluginContainer,C,String,String...)
 - Unknown import "C" in RegisterCommandEvent#register(PluginContainer,C,String,String...)
 - Unknown import "String" in RegisterCommandEvent#register(PluginContainer,C,String,String...)
 - Unknown import "String..." in RegisterCommandEvent#register(PluginContainer,C,String,String...)
Processing: plugin/commands/commandmanager.rst
 - Unknown import "CommandManager" in org.spongepowered.api.command.CommandManager
 - Missing import "CommandManager" in CommandManager
Processing: plugin/commands/index.rst
Processing: plugin/commands/rawcommand.rst
 - Unknown import "PluginContainer" in RegisterCommandEvent#register(PluginContainer,Command,String,String...)
 - Unknown import "Command" in RegisterCommandEvent#register(PluginContainer,Command,String,String...)
 - Unknown import "String" in RegisterCommandEvent#register(PluginContainer,Command,String,String...)
 - Unknown import "String..." in RegisterCommandEvent#register(PluginContainer,Command,String,String...)
````

(There might be some false positives in here, please report them to me except `Import verification error`)

I haven't used the javadoc-imports in a long time so I forgot whether or not they support nested classes via imports and with what syntax or whether you have to qualify them with their root class when using them.

File: source/plugin/commands/argumentparsing.rst (hunk 25 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Interested, i dont get those build errors when building. I can see its wrong in the code, but wasnt notified so never made me look. Ill use your reference for the time being. But is there a specific command to do javadoc building?

Comment (ST-DDT) [reply source/plugin/commands/argumentparsing.rst]: We had to write a custom (experimental) tool for that.

- https://github.com/SpongePowered/SpongeDocs/pull/959
- `cd tools/javadoc-checker && mvn`

Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated the imports. I think I fixed all of them apart from the ``C`` import as thats a generic. Not exactly sure how to do generic parameter imports

Comment (ST-DDT) [reply source/plugin/commands/argumentparsing.rst]: > I think I fixed all of them apart from the C import as thats a generic. Not exactly sure how to do generic parameter imports

IIRC you have to use the actual type instead of the generic name. e.g. `<T extends Actual> void foo(T bar)` -> `foo(Actual foo)`. Just make sure you can actually click on them. Theretically you can go to the jsdocs page and look for the correct value.

File: source/plugin/commands/argumentparsing.rst [hunk 26]
```diff
@@ -0,0 +1,177 @@
+================
+Argument Parsing
+================
+
+.. javadoc-import::
+    org.spongepowered.api.command.parameter.CommandContext
+    org.spongepowered.api.command.parameter.Parameter
+    org.spongepowered.api.command.parameter.Parameter.Builder
```
Comment (ST-DDT) [inline source/plugin/commands/argumentparsing.rst]: org.spongepowered.api.command.parameter.Parameter.Value.Builder

File: source/plugin/commands/argumentparsing.rst [hunk 27]
```diff
@@ -0,0 +1,177 @@
...
+    org.spongepowered.api.command.Command.Builder
+    java.lang.Class
```
Comment (ST-DDT) [inline source/plugin/commands/argumentparsing.rst:11]:
```suggestion
    java.lang.Class
    java.lang.String
```

File: source/plugin/commands/argumentparsing.rst [hunk 28]
```diff
@@ -0,0 +1,177 @@
...
+Many of the parsers may return more than one object; for example, multiple players with a matching username. In that 
+case, you must use the :javadoc:`CommandContext#all(ResourceKey)` method to get the ``Collection`` of possible matches. 
+*Otherwise, the context object will throw an exception!*
+
+When creating a command, consider whether the argument could return multiple values, for example whether a player 
+argument could support multiple players when using a selector. If you support multiple values, the users need to type 
+only one command, and can use an easier command sytax, e.g ``/tell @a Who took the cookies?``
+
+To create a new :javadoc:`Parameter` (argument), use the :javadoc:`Parameter` class that will give you many 
+:javadoc:`Parameter.Builder` options. Each parameter will need its :javadoc:`Parameter.Builder#key(String)`
```
Comment (ST-DDT) [inline source/plugin/commands/argumentparsing.rst]:
```suggestion
:javadoc:`Builder` options. Each parameter will need its :javadoc:`Builder#key(String)` 
```

File: source/plugin/commands/argumentparsing.rst [hunk 29]
```diff
@@ -0,0 +1,177 @@
...
+    org.spongepowered.api.command.Command.Builder
+    java.lang.Class
+
+Minecraft's Brigadier system includes a powerful argument parser that Sponge can take advantage of. 
+It converts the string input to java base types (integer, booleans, string) or game objects 
+(players, worlds, block types, ...). Sponge's system also supports optional arguments and flags. It also handles 
+suggestions of arguments.
+
+The parsed arguments are stored in the :javadoc:`CommandContext` object. If the parser returns a single object, 
+obtain it with :javadoc:`CommandContext#one(ResourceKey)`. Optional and weak arguments may return ``Optional.empty()``
```
Comment (ST-DDT) [inline source/plugin/commands/argumentparsing.rst]: I couldn't find that method. Only when:

ResourceKey -> Parameter.Key

File: source/plugin/commands/argumentparsing.rst [hunk 30]
```diff
@@ -0,0 +1,177 @@
...
+
+Many of the parsers may return more than one object; for example, multiple players with a matching username. In that 
+case, you must use the :javadoc:`CommandContext#all(ResourceKey)` method to get the ``Collection`` of possible matches.
```
Comment (ST-DDT) [inline source/plugin/commands/argumentparsing.rst]: I couldn't find that method. Only when:

ResourceKey -> Parameter.Key

File: source/plugin/commands/argumentparsing.rst [hunk 31]
```diff
@@ -0,0 +1,177 @@
...
+only one command, and can use an easier command sytax, e.g ``/tell @a Who took the cookies?``
+
+To create a new :javadoc:`Parameter` (argument), use the :javadoc:`Parameter` class that will give you many 
+:javadoc:`Parameter.Builder` options. Each parameter will need its :javadoc:`Parameter.Builder#key(String)` 
+filled out before being built. 
+
+Apply the ``Parameter`` to the command builder with the :javadoc:`Command.Builder#addParameter(Parameter)` method. 
+It is possible to pass more than one ``Parameter`` to the method, thus chaining multiple arguments. 
+Example ``/msg <player> <msg>``. This has the same effect as wrapping the ``Parameter`` objects in a 
+:javadoc:`Parameter#seq(Parameter, Parameter, Parameter...)` element.
```
Comment (ST-DDT) [inline source/plugin/commands/argumentparsing.rst]: This link seems to not work at all (javadoc-plugin bug)

**Review by ST-DDT — COMMENTED** [review]
The javadoc plugin is extremely broken.
It probably is incompatible with the JDK version used to generate the javadocs.
I'll stop my review for more javadoc link issues at this.

File: source/plugin/commands/argumentparsing.rst (hunk 25 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Ah ok, then that should be fixed too

File: source/plugin/commands/argumentparsing.rst (hunk 31 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Changed to same method with different parameters

File: source/plugin/commands/argumentparsing.rst (hunk 30 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Changed

File: source/plugin/commands/argumentparsing.rst (hunk 29 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Changed

File: source/plugin/commands/argumentparsing.rst (hunk 28 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 27 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst:11]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 26 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

**Comment by Inscrutable** [comment]
Bump. Some of my review comments seem to have been missed, and are still relevant as far as I can see.

File: source/plugin/commands/commandbuilding.rst (hunk 19 above)
Comment (mosemister) [reply source/plugin/commands/commandbuilding.rst]: Updated

File: source/plugin/commands/commandbuilding.rst (hunk 18 above)
Comment (mosemister) [reply source/plugin/commands/commandbuilding.rst]: Updated

File: source/plugin/commands/commandbuilding.rst (hunk 17 above)
Comment (mosemister) [reply source/plugin/commands/commandbuilding.rst:74]: Updated

File: source/plugin/commands/commandbuilding.rst (hunk 16 above)
Comment (mosemister) [reply source/plugin/commands/commandbuilding.rst]: Updated

File: source/plugin/commands/commandbuilding.rst (hunk 15 above)
Comment (mosemister) [reply source/plugin/commands/commandbuilding.rst]: Updated

File: source/plugin/commands/childcommands.rst (hunk 14 above)
Comment (mosemister) [reply source/plugin/commands/childcommands.rst]: Updated

File: source/plugin/commands/childcommands.rst (hunk 13 above)
Comment (mosemister) [reply source/plugin/commands/childcommands.rst]: Updated

File: source/plugin/commands/childcommands.rst (hunk 12 above)
Comment (mosemister) [reply source/plugin/commands/childcommands.rst]: Updated

File: source/plugin/commands/childcommands.rst (hunk 11 above)
Comment (mosemister) [reply source/plugin/commands/childcommands.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 10 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 9 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 8 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 7 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

File: source/plugin/commands/argumentparsing.rst (hunk 6 above)
Comment (mosemister) [reply source/plugin/commands/argumentparsing.rst]: Updated

**Comment by Inscrutable** [comment]
This can now be merged, unless you have any stray issues @ST-DDT  ?
